docker-compose-developer.yml

```diff
@@ -52,7 +55,7 @@ services:
       MOBILE_APP_QR_INTERVAL_QUEUE: 290000 # 4 minutes and 50 seconds
     volumes:
       - ./grafana-plugin:/etc/app
-      - node_modules_dev:/etc/app/node_modules
+      - ./grafana-plugin/node_modules:/etc/app/node_modules
```

Comment:
When I run `make init`, `node_modules` folder on my host machine is empty and PyCharm cannot correctly understand the project. Or maybe I don't know how to tell PyCharm where to find node_modules?

I can revert these changes if they break developer experience 😄

Reply:
> `node_modules` folder on my host machine is empty

this is correct. [`make init`](https://github.com/grafana/oncall/blob/dev/Makefile#L91) spins up an ephemeral `oncall_ui` container, defined in `docker-compose-dev.yml`, and runs `yarn install && yarn build:dev`.

This installs `node_modules` to the `node_modules_dev` volume mount, not `./grafana-plugin/node_modules` on your host machine & builds the plugin dist to `./grafana-plugin/dist` (on your host-machine). There's more context in the PR description of #1890 on this change. tldr; now you no longer need to have `node` installed on your machine to worry about installing these deps.

Reply:
re: PyCharm, you'd have to point it to the volume mounted `node_modules`, which I suspect is not possible? what if you ran `yarn install` on your host machine (on top of `make init`), does this fix it?

Reply:
> what if you ran yarn install on your host machine (on top of make init), does this fix it?

Yes, that fixes. But in this case there are `node_modules` in `grafana_plugin/node_modules` and in a docker volume. And you're also need installed `node` to run `yarn install` 😄 

> Ideally we store the node_modules only once inside the named volumes.

I only do not understand, why `node_modules` inside the named volume is better than in `grafana_plugin/node_module` on host machine?

Reply:
> why node_modules inside the named volume is better than in grafana_plugin/node_module on host machine

in some sense it's opinionated, but by storing them on the volume, you don't require the developer to have `node` installed on their host machine (because the container takes care of this for you). The frontend will only run w/ specific versions of `node`, so this just makes it one less thing the dev needs to worry about 😄

Reply:
> you don't require the developer to have node installed on their host machine (because the container takes care of this for you).

Exactly the same in this case. You run `make init` and it downloads `node_modules` and exposes them to host machine. So PyCharm can find imports and you can dive into sources of installed dependencies if you need. And I suppose other IDEs also may benefit.

Actually, there is no need to mount  `./grafana-plugin/node_modules:/etc/app/node_modules`, as `./grafana-plugin:/etc/app` already can expose node_modules.

What code editor do you use?

Reply:
lets move this discussion to a separate GitHub issue so that we don't block the other work you've done here 😄 I've created #2097.

Reply:
Sure!